backend/src/main/kotlin/org/loculus/backend/model/ReleasedDataModel.kt

```diff
@@ -85,31 +84,35 @@ open class ReleasedDataModel(
-            ("version" to LongNode(rawProcessedData.version)) +
-            (HEADER_TO_CONNECT_METADATA_AND_SEQUENCES to TextNode(rawProcessedData.submissionId)) +
-            ("accessionVersion" to TextNode(rawProcessedData.displayAccessionVersion())) +
-            ("isRevocation" to BooleanNode.valueOf(rawProcessedData.isRevocation)) +
-            ("submitter" to TextNode(rawProcessedData.submitter)) +
-            ("groupId" to IntNode(rawProcessedData.groupId)) +
-            ("groupName" to TextNode(rawProcessedData.groupName)) +
-            ("submittedDate" to TextNode(rawProcessedData.submittedAtTimestamp.toUtcDateString())) +
-            ("submittedAtTimestamp" to LongNode(rawProcessedData.submittedAtTimestamp.toTimestamp())) +
-            ("releasedAtTimestamp" to LongNode(rawProcessedData.releasedAtTimestamp.toTimestamp())) +
-            ("releasedDate" to TextNode(rawProcessedData.releasedAtTimestamp.toUtcDateString())) +
-            ("versionStatus" to TextNode(versionStatus.name)) +
-            ("dataUseTerms" to TextNode(currentDataUseTerms.type.name)) +
-            ("dataUseTermsRestrictedUntil" to restrictedDataUseTermsUntil) +
-            ("versionComment" to TextNode(rawProcessedData.versionComment))
-
-        if (backendConfig.dataUseTermsUrls != null) {
-            val url = if (rawProcessedData.dataUseTerms == DataUseTerms.Open) {
-                backendConfig.dataUseTermsUrls.open
-            } else {
-                backendConfig.dataUseTermsUrls.restricted
+            mapOf(
+                ("accession" to TextNode(rawProcessedData.accession)),
+                ("version" to LongNode(rawProcessedData.version)),
+                (HEADER_TO_CONNECT_METADATA_AND_SEQUENCES to TextNode(rawProcessedData.submissionId)),
+                ("accessionVersion" to TextNode(rawProcessedData.displayAccessionVersion())),
+                ("isRevocation" to BooleanNode.valueOf(rawProcessedData.isRevocation)),
+                ("submitter" to TextNode(rawProcessedData.submitter)),
+                ("groupId" to IntNode(rawProcessedData.groupId)),
+                ("groupName" to TextNode(rawProcessedData.groupName)),
+                ("submittedDate" to TextNode(rawProcessedData.submittedAtTimestamp.toUtcDateString())),
+                ("submittedAtTimestamp" to LongNode(rawProcessedData.submittedAtTimestamp.toTimestamp())),
+                ("releasedAtTimestamp" to LongNode(rawProcessedData.releasedAtTimestamp.toTimestamp())),
+                ("releasedDate" to TextNode(rawProcessedData.releasedAtTimestamp.toUtcDateString())),
+                ("versionStatus" to TextNode(versionStatus.name)),
+                ("dataUseTerms" to TextNode(currentDataUseTerms.type.name)),
+                ("dataUseTermsRestrictedUntil" to restrictedDataUseTermsUntil),
+                ("versionComment" to TextNode(rawProcessedData.versionComment)),
+            ).let {
+                when (backendConfig.dataUseTermsUrls) {
```

Comment:
This is where the real issue fix is - the rest is mostly refactoring as far as I can tell, oh and extra tests